pkg/controller/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -418,44 +418,65 @@ func compareAndAssignMemberStatuses(reqLogger logr.Logger, toolchainStatus *tool
 	return allOk
 }
 
-// replaceStatusConditions replaces Member status conditions with the new conditions
-func (r *ReconcileToolchainStatus) replaceStatusConditions(status *toolchainv1alpha1.ToolchainStatus, newConditions ...toolchainv1alpha1.Condition) error {
-	// the controller should always update at least the last updated timestamp of the status so the status should be updated regardless of whether
-	// any specific fields were updated. This way a problem with the controller can be indicated if the last updated timestamp was not updated.
+// updateStatusConditions updates ToolchainStatus status conditions with new conditions
+// the controller should always update at least the last updated timestamp of the status so the status should be updated
+// regardless of whether any specific fields were updated. This way a problem with the controller can be indicated if
+// the last updated timestamp was not updated.
+func (r *ReconcileToolchainStatus) updateStatusConditions(logger logr.Logger, status *toolchainv1alpha1.ToolchainStatus,
+	newConditions ...toolchainv1alpha1.Condition) error {
+
+	var updated bool
+
 	var conditionsWithTimestamps []toolchainv1alpha1.Condition
 	for _, newCondition := range newConditions {
 		condition := copyConditionWithTimestampSet(status, newCondition)
 		conditionsWithTimestamps = append(conditionsWithTimestamps, condition)
 	}
-	status.Status.Conditions = conditionsWithTimestamps
-	return r.client.Status().Update(context.TODO(), status)
+
+	status.Status.Conditions, updated = condition.AddOrUpdateStatusConditions(status.Status.Conditions, conditionsWithTimestamps...)
+	if !updated {
+		// Nothing changed
+		logger.Info("ToolchainStatus status conditions unchanged")
+		return nil
+	}
+	logger.Info("updating ToolchainStatus status conditions", "generation", status.Generation, "resource_version", status.ResourceVersion)
+	err := r.client.Status().Update(context.TODO(), status)
+	logger.Info("updated ToolchainStatus status conditions", "generation", status.Generation, "resource_version", status.ResourceVersion)
```

Comment:
Printing generation doesn't make any sense - the spec is empty and generation reflects only the changes in the spec.
I'll remove it

Reply:
removed in e6b496d